src/pcre2_intmodedep.h

```diff
@@ -858,7 +858,7 @@ doing traditional NFA matching (pcre2_match() and friends). */
 
 typedef struct match_block {
   pcre2_memctl memctl;            /* For general use */
-  PCRE2_SIZE heap_limit;          /* As it says */
+  uint32_t heap_limit;            /* As it says */
```

Comment:
PCRE2_SIZE is 64 bit long on 64 bit systems, isn't it?

Reply:
correct, but the heap_limit value can only be lower than UINT_MAX even if it can represent memory sizes larger than 4GB in 64-bit systems because it uses KB for units.